x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/CategoryDefinition.java

```diff
@@ -246,6 +249,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (partitionFieldName != null && partitionFieldValue != null && ReservedFieldNames.isValidFieldName(partitionFieldName)) {
             builder.field(partitionFieldName, partitionFieldValue);
         }
+        builder.field(Result.RESULT_TYPE.getPreferredName(), TYPE.getPreferredName());
```

Comment:
It would be good to add a comment here (or somewhere) to say that `result_type` cannot be relied upon to find category definitions.  We still need to get them by checking for the existence of a `category_id` field until results generated in versions that don't have this are several years old.